### PR TITLE
Move binaries in to the build folder

### LINKED ISSUE #143: Move the netcoreapp2.0 runtime library into the build folder
This way a users application can't actually call into ViewCompilation code.

@Eilon this should be done for RTM

### PR DESCRIPTION
**Pull request by pranavkm**
Maybe fixes #143

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.targets [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+      <ExecArgs>$(ExecArgs) --depsfile &quot;$(ProjectDepsFilePath)&quot;</ExecArgs>
+      <ExecArgs>$(ExecArgs) &quot;$(_MvcViewCompilationBinaryPath)&quot;</ExecArgs>
+      <ExecArgs>$(ExecArgs) @&quot;$(_MvcRazorResponseFilePath)&quot;</ExecArgs>
+    </PropertyGroup>
+
+    <Exec Command="$(ExecArgs)" WorkingDirectory="$(MSBuildProjectDirectory)" />
+  </Target>
+
+  <Target
+    Name="_AddDesktopReferences"
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.targets]: Not super sure if this works - I have to try creating a mismatch in Mvc versions between the project and the tool which I haven't gotten around to as yet.

File: src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.csproj [hunk 2]
```diff
@@ -11,31 +11,14 @@
...
-    <None Include="bin\$(Configuration)\netcoreapp2.0\$(AssemblyName).runtimeconfig.json" Pack="true" PackagePath="lib\netcoreapp2.0\$(AssemblyName).runtimeconfig.json" />
-  </ItemGroup>
-
-  <ItemGroup Condition=" '$(OS)' == 'Windows_NT' ">
-    <None Include="build\net461\*" Pack="true" PackagePath="%(Identity)" />
-    <None Include="bin\$(Configuration)\net461\win7-x64\$(AssemblyName).exe" Pack="true" PackagePath="runtimes\win7-x64\lib\net461\$(AssemblyName).exe" />
-    <None Include="$(X86ProjectDirectory)\bin\$(Configuration)\net461\win7-x86\$(AssemblyName)-x86.exe" Pack="true" PackagePath="runtimes\win7-x86\lib\net461\$(AssemblyName)-x86.exe" />
+    <None Include="build\**\*.*" Pack="true" PackagePath="%(Identity)" />
+    <None Include="$(OutputPath)netcoreapp2.0\$(AssemblyName).dll" Pack="true" PackagePath="build\netstandard2.0\$(AssemblyName).dll" />
+    <None Include="$(OutputPath)net461\win7-x64\$(AssemblyName).exe" Pack="true" PackagePath="build\netstandard2.0\$(AssemblyName)-x64.exe" />
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.csproj]: How will you handle binding redirects? This wont get them auto-generated for you anymore.

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.csproj]: I added the binary as a reference as part of `_AddDesktopReferences`. See my earlier comment

Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.csproj]: oooo that's what you meant there. I see. Ya would be super curious if that works.

Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.csproj]: So when packaging this guy and installing on a local app does it work?

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.csproj]: Not sure what that means?

Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.csproj]: You made it seem like you weren't sure if this worked. After testing, does it work?

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.csproj]: Yup, seems to work. I built Mvc 2.1.0.

```
 <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="Microsoft.AspNetCore.Mvc.Core" publicKeyToken="adb9793829ddae60" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-2.1.0.0" newVersion="2.1.0.0" />
      </dependentAssembly>
    </assemblyBinding>
```

I might be able to hack a fake `Microsoft.AspNetCore.Mvc` package to test this out on an ongoing basis.

File: testapps/ClassLibraryWithPrecompiledViews/ClassLibraryWithPrecompiledViews.csproj [hunk 3]
```diff
@@ -12,6 +12,4 @@
     <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="$(AspNetCoreVersion)" />
   </ItemGroup>
 
-  <Target Name="PrecompileRazorViews" AfterTargets="Build" DependsOnTargets="MvcRazorPrecompile" Condition="'$(TargetFramework)'!=''" />
```
Comment (pranavkm) [inline testapps/ClassLibraryWithPrecompiledViews/ClassLibraryWithPrecompiledViews.csproj]: This fixes https://github.com/aspnet/MvcPrecompilation/issues/117. We explicitly call precompilation on the project from the consuming project.

**Comment by pranavkm** [comment]
New directory structure with this change: 
```
│   Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.nuspec
├───build
│   └───netstandard2.0
│           Microsoft.AspNetCore.Mvc.Razor.ViewCompilation-x64.exe
│           Microsoft.AspNetCore.Mvc.Razor.ViewCompilation-x86.exe
│           Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.dll
│           Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.targets
```

**Comment by NTaylorMullen** [comment]
Ya this is awesome 👍

**Comment by NTaylorMullen** [comment]
Travis looks to be exploding however.

**Comment by pranavkm** [comment]
Travis is sorted - I thought I could remove this - https://github.com/aspnet/MvcPrecompilation/pull/145/files#diff-4d5e7369a237e37020f01b3b255505e0L35 but seemingly not.

**Pull request closed without merging**